### PR TITLE
SONARXML-86 Add ruling

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    .build();
+
+  @Before
+  public void setUp() throws Exception {
+    ProfileGenerator.RulesConfiguration rulesConfiguration = new ProfileGenerator.RulesConfiguration()
+      .add("IllegalTabCheck", "markAll", "true")
+      // FIXME how to setup Template rules?!
+      .add("XPathCheck", "expression", "count(//br)>0")
+      .add("XPathCheck", "message", "Don't do that");
+    Set<String> excludedRules = Collections.singleton("XmlSchemaCheck");
```
Comment (vilchik-elena) [inline its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java]: can be  removed now

File: its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    File litsDifferencesFile = FileLocation.of("target/differences").getFile();
+    SonarScanner build = SonarScanner.create(FileLocation.of("../sources/projects").getFile())
+      .setProjectKey("project")
+      .setProjectName("project")
+      .setProjectVersion("1")
+      .setSourceEncoding("UTF-8")
+      .setSourceDirs(".")
+      .setProperty("dump.old", FileLocation.of("src/test/resources/expected").getFile().getAbsolutePath())
+      .setProperty("dump.new", FileLocation.of("target/actual").getFile().getAbsolutePath())
+      .setProperty("sonar.cpd.skip", "true")
```
Comment (vilchik-elena) [inline its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java]: it's an old property, use `sonar.cpd.exclusions` and `**/*`

File: its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      .setProperty("sonar.analysis.mode", "preview")
+      .setProperty("sonar.issuesReport.html.enable", "true")
```
Comment (vilchik-elena) [inline its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java]: drop line, it's an old prop

File: its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      .setProperty("lits.differences", litsDifferencesFile.getAbsolutePath())
+      .setEnvironmentVariable("SONAR_RUNNER_OPTS", "-Xmx2000m");
```
Comment (vilchik-elena) [inline its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java]: do we need this one?

File: its/sources/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Github projects:
```
Comment (vilchik-elena) [inline its/sources/README.md]: wow! great!

File: its/sources/sonar-project.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+sonar.projectKey=xml-ruling
```
Comment (vilchik-elena) [inline its/sources/sonar-project.properties]: why this file? you are setting everything in the test

File: its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java (hunk 4 above)
Comment (m-g-sonar) [reply its/ruling/src/test/java/org/sonarsource/xml/it/XmlRulingTest.java]: probably not, copy pasted from python.

File: its/sources/sonar-project.properties (hunk 6 above)
Comment (m-g-sonar) [reply its/sources/sonar-project.properties]: For manual testing on local instance, no need to configure anything. just run sonar scanner from there.

File: its/plugin/pom.xml [hunk 7]
```diff
@@ -9,9 +9,9 @@
     <version>1.6.0-SNAPSHOT</version>
   </parent>
 
-  <artifactId>it-xml</artifactId>
+  <artifactId>it-xml-plugin</artifactId>
 
-  <name>SonarXML :: IT</name>
+  <name>SonarXML :: IT Plugin</name>
```
Comment (vilchik-elena) [inline its/plugin/pom.xml]: remove orchestator version in this file

Comment (m-g-sonar) [reply its/plugin/pom.xml]: Let's do that in another PR, as I would also like to update version of SQ WebService API which is used

**Pull request merged**